components/dashboard-modules/SunnyTimeframe.tsx

```diff
@@ -11,12 +11,14 @@ import { hoursToMinutes, millisecondsToHours } from 'date-fns';
 import useDateFormatter from '~/lib/hooks/useDateFormatter';
 
 const SunnyTimeframe: FC<{ siteUUID: string }> = ({ siteUUID }) => {
-  const { forecastData } = useSiteData(siteUUID);
+  const { forecastData, isLoading } = useSiteData(siteUUID);
   const [isRelativeTime, setIsRelativeTime] = useState(false);
   const { timeFormatter } = useDateFormatter(siteUUID);
 
   if (!forecastData) {
-    return <NumberDisplay title="Loading" value="..." />;
+    return (
+      <NumberDisplay title="Loading" value="Loading..." isLoading={isLoading} />
+    );
```

Comment:
Though it's hard to tell, you may have noticed that the skeleton effect does not fade out on the `SunnyTimeframe` element, like it does for the other `<NumberDisplay` components on the dashboard. The reason for this is slightly complicated, but it comes down to these lines. Since this component returns early if the `forecastData` doesn't exist, this instance of `<NumberDisplay` will initially be rendered with the skeleton effect. Once the data does exist, this instance will disappear and the one further down will be rendered instead. Despite them being the same component, their rendered HTML elements are different (note that the one below has an `onClick`, which causes the component to show it as a button), and when elements are swapped out on the page they lose their ability to transition CSS wise.

There's a couple ways we could fix this:
- Remove the `onClick` prop for `<NumberDisplay` and don't let them appear as buttons at all (will prevent HTML element swap)
- Remove this return with a `<NumberDisplay` specifically for loading. Instead, move the logic to compute the sunny text and value to another function, and have it consider the case where `forecastData` is undefined, and just return text with "loading". Then, you can always render the bottom `<NumberDisplay` with its `onClick`
- Just add an `onClick` prop to this instance but have it do nothing (`onClick={() => {}}`)

For now, let's go with the last option. Do you think another option is a better long term solution though?

Reply:
Just committed my changes for the last option. I think that maybe the second option would be a better long term solution.